src/OrleansRuntime/Placement/IPlacementContext.cs

```diff
@@ -54,22 +56,32 @@ internal interface IPlacementContext
         /// <returns></returns>
         bool TryGetActivationData(ActivationId id, out ActivationData activationData);
 
-        void GetGrainTypeInfo(int typeCode, out string grainClass, out PlacementStrategy placement, string genericArguments = null);
+        void GetGrainTypeInfo(int typeCode, out string grainClass, out PlacementStrategy placement, out MultiClusterRegistrationStrategy strategy, string genericArguments = null);
     }
 
     internal static class PlacementContextExtensions
     {
-        public static Task<List<ActivationAddress>> Lookup(this IPlacementContext @this, GrainId grainId)
+        public static async Task<List<ActivationAddress>> Lookup(this IPlacementContext @this, GrainId grainId)
         {
             List<ActivationAddress> l;
-            return @this.FastLookup(grainId, out l) ? Task.FromResult(l) : @this.FullLookup(grainId);
+            if (@this.FastLookup(grainId, out l))
+            {
+                return l;
+            }
+            else
+            {
+                var result = await @this.FullLookup(grainId);
```

Comment:
Fixed: it no longer does an extra await. This await has disappeared as a result of a refactoring, which makes the return type of all grain directory lookups to be a struct containing a list of addresses and a version tag  (a change suggested by @jdom).